Qt/Qt_Main.cpp

```diff
@@ -3077,6 +3077,13 @@ bool Control2Pressed(void){
 }
 */
 
+Qt::KeyboardModifier HorizontalModifier() {
+  if (SETTINGS_read_bool("alt_as_horizonal_scroll_modifier", false))
```

Comment:
This is actually very inefficient. Pretty sure it's reading the whole configuration file every you press a key, or at least very often.

Reply:
It's probably easy to fix the SETTINGS_* functions to cache all content.

Reply:
It's actually worse. It's reading the settings file every time the mouse cursor is moved while pressing a mouse button. Simple to fix though. Going to push something like this:

```
kjetil@tthp:~/radium$ git diff common/settings.cpp
diff --git a/common/settings.cpp b/common/settings.cpp
index 4cb4cf26b..f8f1d69f3 100644
--- a/common/settings.cpp
+++ b/common/settings.cpp
@@ -326,6 +326,8 @@ static void SETTINGS_put(const char* key, QString val){
 
 // Warning, called before GC_init, so it cannot allocate with talloc or talloc_atomic.
 static QString SETTINGS_get(const char* key){
+  printf("SEtTING_get called for \"%s\"\n", key);
+
   QVector<QString> lines = get_lines(key);
   if(lines.size()==0)
     return not_found;
@@ -358,8 +360,18 @@ void SETTINGS_unset_custom_configfile(void){
   custom_configuration_filename="";
 }
 
+static QHash<QString, bool> g_bools;
+
 bool SETTINGS_read_bool(const char* key, bool def){
-  return SETTINGS_read_string(key, def==true?"true":"false")[0] == 't';
+  auto it = g_bools.find(key);
+  if (it != g_bools.end())
+    return it.value();
+
+  bool ret = SETTINGS_read_string(key, def==true?"true":"false")[0] == 't';
+
+  g_bools[key] = ret;
+
+  return ret;
 }
 
 int64_t SETTINGS_read_int(const char* key, int64_t def){
@@ -434,6 +446,7 @@ vector_t *SETTINGS_get_all_lines_starting_with(const char *prefix){
 }
 
 void SETTINGS_write_bool(const char* key, bool val){
+  g_bools[key] = val;
   SETTINGS_write_string(key, val==true?"true":"false");
 }
```

Reply:
> This is actually very inefficient. Pretty sure it's reading the whole configuration file every you press a key, or at least very often.

Oh, I apologize, that's my fault. I didn't check how the SETTINGS_* functions works. I simply assumed that the values were cached in a map underneath rather than directly loaded from the configuration file (I thought they were an interface to the settings in memory). Now that I think about it, we could simply add a boolean variable and load it at the start of the application and settings update. I'm not sure if it will be faster or worth to do it now when we have caching mechanism